tests/test_analysis_assurance_depth_and_graph_overlap.py

```
@@ -457,3 +457,51 @@ def test_unit_level_isdisjoint_vs_inequality_direct(self, tmp_path: Path) -> Non
+        # decl comes back genuinely empty once the fix is in place.
+        exports = [
+            "_ZNSt6vectorIiSaIiEEC1Ev",
+            "_ZN3tbb6detail2r13fooEv",
+            "_private_c",
+        ]
+        old, new = _header_pair()
+        for snap in (old, new):
+            surface = link_source_abi([SourceAbiTu()], exported_symbols=exports)
+            assert surface.unmatched["symbols_without_decl"] == []
+            snap.build_source = BuildSourcePack(
+                root=Path("/tmp/nonexistent-pack-no-double-count"),
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/nonexistent-pack-no-double-count"

(S108)